cheerio-scraper/src/crawler_setup.js

```diff
@@ -154,18 +160,25 @@ class CrawlerSetup {
             requestOptions: {
                 headers: {},
             },
+            useSessionPool: true,
+            persistCookiesPerSession: true,
+            sessionPoolOptions: {
+                sessionOptions: {
+                    maxUsageCount: this.maxSessionUsageCount
+                }
+            }
         };
 
-        if (this.cookieJar) {
-            options.requestOptions.cookieJar = this.cookieJar;
+        if (this.input.proxyRotation === 'UNTIL-FAILURE') {
```

Comment:
Please use constants, not hard-coded strings.

Reply:
Even if it's not used anywhere else? I find having `const UNTIL_FAILURE = 'UNTIL_FAILURE'` on line 10 and then on line 172 `=== UNTIL_FAILURE` quite redundant. It would make sense if we could inject the consts into the INPUT_SCHEMA, but that would require a build process.

```js
const timeout = secs * 1000;

// vs
const THOUSAND = 1000;
// ...
const timeout = secs * THOUSAND;
```

I really don't know...

Reply:
It is a good habit to avoid hard-coding constants into the code, and we are following it across our entire code base. The point is, using "Find usage" tool in your IDE you can easily see where the constants is used, if you want to change the constant you can do it in one place... well, INPUT_SCHEMA.json is bit unfortunate here but at least, you can add a comment to the constant definition saying this needs to match the INPUT_SCHEMA.json. In this case, we could have:

But really the most important thing is maintaining consistency how we write code across our entire code base, so that people can easily switch between projects and know how things are done and how to do things.

Reply:
https://github.com/apifytech/actor-scraper/commit/456ca39fade2a248fc61a06293a1014bbae8977f